importer/src/path_canonicalizer.rs

```diff
@@ -925,6 +954,28 @@ fn check_type_declaration(
         .try_for_each(|ty| check_type(location, ty, state, &type_vars, &Default::default()))
 }
 
+fn check_struct_declaration(
+    location: &AbsoluteSymbolPath,
+    struct_decl: &StructDeclaration<Expression>,
+    state: &mut State<'_>,
+) -> Result<(), Error> {
+    struct_decl
+        .fields
+        .iter()
+        .try_fold(BTreeSet::default(), |mut acc, (name, _)| {
+            acc.insert(name.clone())
+                .then_some(acc)
+                .ok_or(format!("Duplicate field `{name}` in struct `{location}`"))
```

Comment:
Are you doing the same in statement processor? I think it should be done there as well.